timeseries/src/autogluon/timeseries/trainer/ensemble_composer.py

```diff
@@ -19,33 +20,85 @@
 
 
 class EnsembleComposer:
-    """Helper class for TimeSeriesTrainer to build multi-layer stack ensembles."""
+    """Helper class for TimeSeriesTrainer to build multi-layer stack ensembles.
+
+    This class depends on the trainer to provide the necessary initialization parameters, training
+    and validation data, as well as having fit the base (non-ensemble) models and persisted their
+    out-of-fold predictions which will be used for ensemble training.
+
+    Parameters
+    ----------
+    path
+        Path of the calling TimeSeriesTrainer. EnsembleComposer finds the model objects and their
+        out-of-fold prediction artifacts with respect to this path. EnsembleComposer only saves
+        ensemble models and their out-of-fold predictions to this folder (i.e., does not pickle
+        itself).
+    prediction_length
+        Number of time steps to forecast.
+    eval_metric
+        Metric used to evaluate ensemble performance.
+    target
+        Name of the target column in the time series data.
+    num_windows_per_layer
+        Number of windows used for training each ensemble layer. Length must match the number of layers
+        in ensemble_hyperparameters. Example: (3, 2) means first layer uses 3 windows, second layer uses
+        2 windows. See notes below for further explanation.
+
+        Base models must have OOF predictions saved for all sum(num_windows_per_layer) windows, prior
+        to this class being called.
+    ensemble_hyperparameters
+        Ensemble configuration. A list of dicts, one per layer. If an ensemble model should be fitted
+        with multiple hyperparameter configurations, a list of dicts may be provided as the value.
+        Each layer's dict maps ensemble names to either a single hyperparameter dict or a list of
+        hyperparameter dicts.
+
+        Examples:
+        - ``[{"GreedyEnsemble": {}}, {"GreedyEnsemble": {}}]`` for 2 layers of greedy ensembles.
+        - ``[{"GreedyEnsemble": [{"ensemble_size": 10}, {"ensemble_size": 20}]}]`` for a single layer of
```

Comment:
How will the models be named currently with such a config? `GreedyEnsemble` and `GreedyEnsemble_2` or something else?

Regarding the discussion point in the PR description: I think that `GreedyEnsemble_L2` and `GreedyEnsemble_L3` would be a good and consistent way to label L2/L3 ensemble models, but I'm not sure how well this plays with the `_2`, `_3`, ... suffixes for different variants of the same model. I also see that we cannot force "single config per ensemble class in a level" because of the different variants of the `LinearEnsemble` that we might want to combine.